datastore/longhorn.go

```diff
@@ -124,10 +124,15 @@ func (s *DataStore) isSettingValueChanged(name types.SettingName, value string)
 		}
 		return true
 	}
-	if setting.Value == value {
-		return false
+
+	// Also check the setting definition.
+	definition, ok := types.GetSettingDefinition(name)
+	if !ok {
+		logrus.Errorf("customized default setting %v is not defined", name)
+		return true
 	}
-	return true
+
+	return (setting.Value != value || definition.Default != value)
```

Comment:
This function checks if a customized value can be set as the definition default value. Then I am thinking if `setting.Value != value`. BTW, we may need to change the function name as well.

Reply:
The function checks whether a change would have any effect or can be skipped.  The logic for changing the configmap default definition also propagates the change to the global setting, regardless of its current value.  To decide that there is a change to be made, it needs to compare to both, because the global setting might be different from the default definition.

I agree that the function could be better named, so I've pushed that improvement.

Reply:
NIT: The new naming is confusing as well... I am thinking if we can move the config map resource version check into this function and rename this function to `shouldApplyCustomizedSettingValue`

Reply:
`shouldApplyCustomizedSettingValue` it shall be.
For the other, are you asking for the check of config map resource version in the setting's Annotations that is currently done in `syncSettingCRsWithCustomizedDefaultSettings` to be moved to `shouldApplyCustomizedSettingValue`?
Done.

Reply:
I think I'm caught up with the feedback items.  Can you check and approve, @shuo-wu?  Thanks!